### ISSUE TITLE
This happened when I said 'yes' to optimizing KataGo

### ISSUE THREAD

**Issue by B7ADE-RUNNER** [wonsiks/BadukMegapack#30]
System Processor: AMD RYZEN 5 3550H MOBILE PROCESSOR WITH RADEON VEGA 8 Graphics.

I downloaded the installer from the GoogleDrive Link. I initiated the installer. The first prompt asking about the GPU model. I chose option (3), although I think the graphics for my system is processed on the CPU. It says for the hardware specifications at the AMD website: # of CPU cores - 4, # of Threads 8, # of GPU cores - 8. This to me makes me think option (4), CPU embedded GPU, may have been the other option. My first issue is can you confirm a correct or incorrect choice, here? Which number should I have selected.

The command prompt came up and everything went automatic until it asked me if I want to optimize KataGo. I said yes because it was recommend and then it gave this error:

The syntax of the command is incorrect. File default_gtp.cfg already exists. Overwrite it and create a new configuration file? (y/n)

I put 'n' because I did not want to overwrite anything. I wrongly assumed it would leave the installer and not carry the processes forward. 

Whenever I start Lizzie, this is the error I receive:

Engine process ended unintentionally for some reason. You may find more information in the GTP console. Engine command: ./katago.exe gtp - model KataGo20b.gz

I get the same errors for KataGo15b.gz and KataGo10b.gz.

I wanted to use this installer because I am not good with computers. Any help would be appreciated, thank you so much for your time.

**Comment by B7ADE-RUNNER** [comment]
Also am getting errors, Failed to estimate score, Did not find "YAZenGtp.exe" or "Zen.d11"!

**Comment by GopherAl** [comment]
I think you have made an error or two. I've tried BadukMegapack in the past and had issues with it. 

Currently LizGoban is the simplest to set up and is easy to install for anyone. It is very easy to use and has lots of good features. I like it the best. 

KaTrain is also easy to use and install. It has many options and is popular.

Both of these programs are available on GitHub. I'm away from home so it's not easy to give more info. Good luck and happy go!

**Comment by GopherAl** [comment]
Both of those programs focus on KataGo

**Comment by hope366** [comment]
The details of the error may be displayed in the GTP console. Can you paste it?

**Comment by GopherAl** [comment]
Clarification: Initially I had trouble getting KataGo running nicely in LizGoban but that was a long time ago. Nowadays all you have to do is download it and install it, both of which are simple to do. The same is true of KaTrain. Both of these are excellent options for playing KataGo.

**Comment by GopherAl** [comment]
I used BadukMegapack for similar reasons to you so that I could play KataGo. These days it's much easier to install things yourself. Using BadukMegapack just complicates things and causes problems It seems to me that you chose many incorrect options when using BadukMegapack. LizGoban and KaTrain don't require any set up; they do that automatically. And at the moment they are both more developed and better supported than Lizzie.

**Comment by B7ADE-RUNNER** [comment]
@hope366 

> The details of the error may be displayed in the GTP console. Can you paste it?

This is what it says when I start Lizzie with KataGo

Uncaught exception: Key 'startupPrintMessageToStderr' + was specified multiple times in D:\igo\lizzie/default_gtp.cfg, you probably didn't mean to do this, please delete one of them

Also, I found the default_gtp.cfg file in the lizzie folder.

**Comment by B7ADE-RUNNER** [comment]
> I used BadukMegapack for similar reasons to you so that I could play KataGo. These days it's much easier to install things yourself. Using BadukMegapack just complicates things and causes problems It seems to me that you chose many incorrect options when using BadukMegapack. LizGoban and KaTrain don't require any set up; they do that automatically. And at the moment they are both more developed and better supported than Lizzie.

Yeah, I was going to install them separately. It looked fairly easy, but I thought why not try this if I can get them all within a nifty install program. Everything seems to be working fine with these programs if I just commit to loading in the LeelaZero engines and not use KataGo at all. But I may just uninstall the files, clean things up a bit, and then install all the programs separately. There are pros and cons to all of them and it is nice to play around with different programs, so that I can let my whims keep my interest in this game alive.

**Comment by B7ADE-RUNNER** [comment]
@hope366 

So, it is accepting the KataGo engine now. 

> Also am getting errors, Failed to estimate score, Did not find "YAZenGtp.exe" or "Zen.d11"!

This is still an issue. I do see the YAZenGtp.exe file. I double click it and a terminal window pops up and closes instantly.

**Comment by hope366** [comment]
@B7ADE-RUNNER
With the previous lizzie, I think users were often annoyed when the lizzie failed to start for some reason and the GTP console didn't show up.
The recent improvements in lizzie have made it much easier to use because in most cases, lizzie will report where the problem is.
In this case, it seems to be due to the fact that "startupPrintMessageToStderr = true" is specified more than once in the default_gtp.cgf. 
I don't know for what reason it was specified more than once, but if you remove the extra part, it should start up fine.
Open default_gtp.cgf. The "startupPrintMessageToStderr = true" is usually found in the "# Logs and files" column at the top of the file, but since the megapack makes an additional specification at the end of the file, it's probably the same in both of these locations . If you delete one or the other, the error will be gone.

For the errors related to "estimate score", you must have both YAZenGtp.exe and Zen.dll in the lizzie folder to get them to work.
Usually you have YAZenGtp.exe, but you have to provide Zen.dll yourself. Since this is a commercial program, I think it is intended for "Zenith Go" users.
I have tried this feature, but it wasn't as good as I thought it would be. So I don't think it's a problem to ignore this feature.
Of course if you can use this feature in some way that would be great.

**Comment by B7ADE-RUNNER** [comment]
> @B7ADE-RUNNER
> With the previous lizzie, I think users were often annoyed when the lizzie failed to start for some reason and the GTP console didn't show up.
> The recent improvements in lizzie have made it much easier to use because in most cases, lizzie will report where the problem is.
> In this case, it seems to be due to the fact that "startupPrintMessageToStderr = true" is specified more than once in the default_gtp.cgf.
> I don't know for what reason it was specified more than once, but if you remove the extra part, it should start up fine.
> Open default_gtp.cgf. The "startupPrintMessageToStderr = true" is usually found in the "# Logs and files" column at the top of the file, but since the megapack makes an additional specification at the end of the file, it's probably the same in both of these locations . If you delete one or the other, the error will be gone.
> 
> For the errors related to "estimate score", you must have both YAZenGtp.exe and Zen.dll in the lizzie folder to get them to work.
> Usually you have YAZenGtp.exe, but you have to provide Zen.dll yourself. Since this is a commercial program, I think it is intended for "Zenith Go" users.
> I have tried this feature, but it wasn't as good as I thought it would be. So I don't think it's a problem to ignore this feature.
> Of course if you can use this feature in some way that would be great.

@hope366 
Well, you are very informative and helpful. I was able to fix the first issue using your guidance, so great, and the second issue, I don't even recognize 'zenith go' as being part of this suite. KataGo seems to put squares of varying opacity on the lattice to indicate potential territory for each side - this is really terrific. 



Actually, just searching for the zen.dll can take one to the github page for lizzie, where that file is. I did not even have to update the other one. Like you said it needed both. For whatever reason, that file did not get installed, but if I copy that other one in the lizzie directory the thing works as it is supposed to. Who knows what other problems await, but it seems to work as intended for now. Some of the engines like Kata 40 or anything > 20 are not loading into the programs, but I mean for now, things seem okay.

**Comment by wonsiks** [comment]
@B7ADE-RUNNER Sorry for late reply.
1) During installing the pack Katago tuning step will overwrite default config file and if user doesn't let it be overwritten then things would go wrong(duplication will be occur). Because I put some additional settings which are not contained in automatically created config file.
2) zen.dll is a part of commercial product so I didn't put it in the pack. However, I put download_zendll.bat file in the lizzie folder with which you can download zen.dll automatically.

**Comment by B7ADE-RUNNER** [comment]
@wonsiks 

1. I wound up letting it overwrite via the very convenient 'Generate KataGo' config. After I let it overwrite, something I should have done to begin with, KataGo engines started working with the programs. 

2. I wound up downloading zen.dll directly from the lizzie github page. I did not recognize this link at first. 

Thank you, everything worked out.
